Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -126,6 +219,24 @@ MediaSource::~MediaSource()
     ASSERT(isClosed());
 }
 
+#if !RELEASE_LOG_DISABLED
+Ref<Logger> MediaSource::logger(ScriptExecutionContext& context)
+{
+    if (RefPtr document = dynamicDowncast<Document>(context))
+        return document->logger();
+
+    Ref logger = Logger::create(this);
+    logger->addObserver(*this);
+    return logger;
+}
+
+void MediaSource::didLogMessage(const WTFLogChannel&, WTFLogLevel, Vector<JSONLogValue>&&)
+{
+    // FIXME: Add logging for when MediaSource is running in worker.
+}
```

Comment:
I think you can do something like: 
```
scriptExecutionContext()->postTaskToResponsibleDocument([channel, level, log] (Document& document) {
    document->didLogMessage(channel, level, WTFMove(log));
});
```

Reply:
didLogMessage is a private member of Document

Reply:
Document logger can probably be used for the MediaSource as it seems to be thread safe. This seems best be done as a follow-up.